### PR TITLE
WIP: Add an endpoint for heap dumps

### PR DESCRIPTION
**Pull request by mrunalp**
This is to aid inspecting memory usage

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This PR adds support to get crio heap dumps for inspecting memory usage.

#### Which issue(s) this PR fixes:
None

#### Does this PR introduce a user-facing change?

```release-note
Add support for gathering heap dumps
```

### CODE REVIEW

File: server/inspect.go [hunk 1]
```diff
@@ -186,5 +191,25 @@ func (s *Server) GetInfoMux(enableProfile bool) *bone.Mux {
 		mux.Get("/debug/pprof/*", http.HandlerFunc(pprof.Index))
 	}
 
+	if enableProfile {
+		mux.Get("/debug/dumpheap", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			dumpFilePath := filepath.Join("/tmp", fmt.Sprintf(
+				"crio-heapdump-%s.out",
```
Comment (saschagrunert) [inline server/inspect.go:197]: Idea: Add a subcommand to `crio-status` to dump to a specific file location.

**Comment by saschagrunert** [comment]
A friendly reminder that this PR had no activity for 30 days.

**Comment by haircommander** [comment]
do we still want this @mrunalp @saschagrunert

**Comment by saschagrunert** [comment]
> do we still want this @mrunalp @saschagrunert

Yep, I think so! :blush:

**Comment by sohankunkerkar** [comment]
@cri-o/cri-o-maintainers We still require this. Can we consider opening it as a "good-first-issue" to encourage external contributions?

**Comment by haircommander** [comment]
good idea!

**Comment by sohankunkerkar** [comment]
Closing this in favor of https://github.com/cri-o/cri-o/issues/7307

**Pull request closed without merging**